### PR TITLE
vectorscan: Submission

### PR DESCRIPTION
**Pull request by essandess**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y / Command Line Tools x.y.z

###### Verification 
Have you

- [ ] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: textproc/vectorscan/Portfile [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                    -DARCH_X86_64=On \
+                    -DBUILD_AVX512VBMI=On
+    }
+    default {
+        known_fail  yes
+        pre-fetch {
+            ui_error "${subport} @ ${version} only supported for architectures ${supported_archs}"
+            return -code error "Unsupported architecture: ${build_arch}"
+        }
+    }
```
Comment (ryandesign) [inline textproc/vectorscan/Portfile:78]: This shouldn't be necessary. MacPorts already checks `supported_archs` for you and presents a suitable error message when needed.
```suggestion
```

File: textproc/hyperscan/Portfile [hunk 2]
```diff
@@ -37,6 +37,8 @@ github.tarball_from archive
 # https://github.com/intel/hyperscan/issues/197
 supported_archs     x86_64
 
+conflicts           vectorscan
+
 # Use ${python.default_version}
 # name consistency with ${prefix}/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/_resources/port1.0/group/python-1.0.tcl
 set python_default_version 311
```
Comment (reneeotten) [inline textproc/hyperscan/Portfile:44]: this should be using Python 3.12 if possible

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: textproc/vectorscan/Portfile [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+# strnlen
+legacysupport.newest_darwin_requires_legacy 10
+
+github.setup        VectorCamp vectorscan 5.4.11 vectorscan/
+revision            0
+categories          textproc
+license             BSD
+maintainers         nomaintainer
+github.tarball_from archive
+supported_archs     arm64 x86_64
```
Comment (barracuda156) [inline textproc/vectorscan/Portfile:18]: @essandess Do we know why a “portable” fork fails to be portable?

Comment (ryandesign) [reply textproc/vectorscan/Portfile:18]: It's more portable than hyperscan which only supported x86_64. Per the readme:

> Currently ARM NEON/ASIMD and Power VSX are 100% functional. ARM SVE2 support is in ongoing with access to hardware now. More platforms will follow in the future. Further more, starting 5.4.12 there is now a SIMDe port, which can be either used for platforms without official SIMD support, as SIMDe can emulate SIMD instructions

Comment (barracuda156) [reply textproc/vectorscan/Portfile:18]: Thank you, interesting. Possibly, next version will work on powerpc.

P. S. Perhaps of little interest for Macports, but “Power VSX are 100% functional” implies that it should work on ppc64, though on cpus with ISA 2.07 (or something around that), so macOS won’t, but BSD and Linux on supported hardware may.

File: textproc/hyperscan/Portfile (hunk 2 above)
Comment (essandess) [reply textproc/hyperscan/Portfile]: Thank you. Fixed.

File: textproc/vectorscan/Portfile (hunk 1 above)
Comment (essandess) [reply textproc/vectorscan/Portfile]: Thank you. Fixed.

**Comment by essandess** [comment]
I don't observe this CI error:

```sh
2024-04-12T09:30:42.9836840Z [ 24%] Generating libstemmer/stem_UTF_8_arabic.c
2024-04-12T09:30:42.9837260Z [ 25%] Generating libstemmer/stem_UTF_8_turkish.c
…
2024-04-12T09:30:42.9848100Z cd /opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_mail_rspamd/rspamd/work/build/contrib/snowball && env ASAN_OPTIONS=detect_leaks=0 /opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_mail_rspamd/rspamd/work/build/contrib/snowball/snowball /opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_mail_rspamd/rspamd/work/rspamd-3.8.4/contrib/snowball/algorithms/turkish.sbl -o /opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_mail_rspamd/rspamd/work/build/contrib/snowball/libstemmer/stem_UTF_8_turkish -eprefix turkish_UTF_8_ -r ../runtime -u
2024-04-12T09:30:42.9913270Z /bin/sh: line 1: 25945 Killed: 9               env ASAN_OPTIONS=detect_leaks=0 /opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_mail_rspamd/rspamd/work/build/contrib/snowball/snowball /opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_mail_rspamd/rspamd/work/rspamd-3.8.4/contrib/snowball/algorithms/arabic.sbl -o /opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_mail_rspamd/rspamd/work/build/contrib/snowball/libstemmer/stem_UTF_8_arabic -eprefix arabic_UTF_8_ -r ../runtime -u
2024-04-12T09:30:42.9917850Z /bin/sh: line 1: 25944 Killed: 9               env ASAN_OPTIONS=detect_leaks=0 /opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_mail_rspamd/rspamd/work/build/contrib/snowball/snowball /opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_mail_rspamd/rspamd/work/rspamd-3.8.4/contrib/snowball/algorithms/turkish.sbl -o /opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_mail_rspamd/rspamd/work/build/contrib/snowball/libstemmer/stem_UTF_8_turkish -eprefix turkish_UTF_8_ -r ../runtime -u
2024-04-12T09:30:42.9920090Z make[2]: *** [contrib/snowball/libstemmer/stem_UTF_8_turkish.c] Error 137
2024-04-12T09:30:42.9920470Z make[2]: *** Waiting for unfinished jobs....
2024-04-12T09:30:42.9920840Z make[2]: *** [contrib/snowball/libstemmer/stem_UTF_8_arabic.c] Error 137
2024-04-12T09:30:42.9921810Z make[2]: Leaving directory `/opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_mail_rspamd/rspamd/work/build'
2024-04-12T09:30:42.9922510Z make[1]: *** [contrib/snowball/CMakeFiles/stemmer.dir/all] Error 2
```

**Comment by reneeotten** [comment]
Feel free to reopen the PR once you had time to fix this properly.

**Comment by reneeotten** [comment]
> I'm running working binaries of `rspamd`+`vectorscan` on `x86_64`.

The failure on mac-os14 is on arm. Please verify locally that this works for you as well. 

You cannot claim every time that the CI is the problem and not your changes.

**Pull request closed without merging**